#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      maxContactJoints 50
    }
  ]
}
Viewpoint {
  orientation -0.16895044469060289 0.3339892883892655 0.9273116533723059 0.9576464436661775
  position -0.39163339747548637 0.543080732849777 0.26759349169446683
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 3
}
Robot {
  translation -4.97e-06 1.06 0.17134
  rotation 0 0 1 4.692820414042842e-06
  children [
    DEF trunk_solid Solid {
      children [
        DEF trunk CadShape {
          url [
            "../openduck_in_solvespace/split_parts/part_1.obj"
          ]
        }
        DEF left_motor_1 CadShape {
          url [
            "../openduck_in_solvespace/split_parts/part_5.obj"
          ]
        }
        DEF right_motor_1 CadShape {
          url [
            "../openduck_in_solvespace/split_parts/part_4.obj"
          ]
        }
      ]
      name "trunk_solid"
      boundingObject DEF trunk_shape Group {
        children [
          DEF trunk_1 Pose {
            translation 0.035 0 0.0035
            children [
              Box {
                size 0.108 0.038 0.007
              }
            ]
          }
          DEF trunk_2 Pose {
            translation 0.035 0 0.036
            children [
              Box {
                size 0.025 0.038 0.058
              }
            ]
          }
          DEF trunk_3 Pose {
            translation 0.035 0.005 0.07
            children [
              Box {
                size 0.044 0.08 0.01
              }
            ]
          }
          DEF trunk_4 Pose {
            translation 0.05375 0.005 0.0885
            children [
              Box {
                size 0.0065 0.08 0.027
              }
            ]
          }
          DEF trunk_5 Pose {
            translation 0.01625 0.005 0.0885
            children [
              Box {
                size 0.0065 0.08 0.027
              }
            ]
          }
          DEF left_motor1 Pose {
            translation 0 0.01211 0.0749197
            children [
              Box {
                size 0.02472 0.04522 0.029
              }
            ]
          }
          DEF right_motor1 Pose {
            translation 0.07 0.0121 0.0749
            rotation 0 1 0 4.692820414042842e-06
            children [
              Box {
                size 0.02472 0.04522 0.029
              }
            ]
          }
        ]
      }
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.07 0 0.05222
      }
      device [
        RotationalMotor {
          name "t1"
          maxVelocity 300
          maxTorque 300
        }
        PositionSensor {
          name "t1_sensor"
        }
      ]
      endPoint DEF left_second_motor Solid {
        translation -4.163336342344337e-17 3.469446951953614e-18 0
        children [
          DEF left_second_motor CadShape {
            url [
              "../openduck_in_solvespace/split_parts/part_2.obj"
            ]
          }
          DEF left_shoulder HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 1 0
              anchor 0.07 -0.0145 0.0175
            }
            device [
              RotationalMotor {
                name "t2"
                maxVelocity 300
                maxTorque 300
              }
              PositionSensor {
                name "t2_sensor"
              }
            ]
            endPoint DEF left_shoulder Solid {
              translation 5.551115123125783e-17 -2.0816681711721685e-17 6.938893903907228e-18
              rotation 0 1 0 0
              children [
                DEF left_shoulder CadShape {
                  url [
                    "../openduck_in_solvespace/split_parts/part_6.obj"
                  ]
                }
                DEF left_third_motor CadShape {
                  url [
                    "../openduck_in_solvespace/split_parts/part_8.obj"
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    anchor 0.15313 0.01464 0.01652
                  }
                  device [
                    RotationalMotor {
                      name "t3"
                      maxVelocity 300
                      maxTorque 300
                    }
                    PositionSensor {
                      name "t3_sensor"
                    }
                  ]
                  endPoint DEF third_motor_leg Solid {
                    translation 1.1102230246251565e-16 5.204170427930421e-18 3.469446951953614e-18
                    rotation 1 0 0 0
                    children [
                      DEF left_leg_1 CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_10.obj"
                        ]
                      }
                      DEF left_leg_2 CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_11.obj"
                        ]
                      }
                      DEF left_fourth_motor CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_14.obj"
                        ]
                      }
                      DEF fourth_motor HingeJoint {
                        jointParameters HingeJointParameters {
                          anchor 0.15413 0.01628 -0.06175
                        }
                        device [
                          RotationalMotor {
                            name "t4"
                            maxVelocity 300
                            maxTorque 300
                          }
                          PositionSensor {
                            name "t4_sensor"
                          }
                        ]
                        endPoint DEF fourth_motor_leg Solid {
                          rotation 1 0 0 0
                          children [
                            DEF left_leg_3 CadShape {
                              url [
                                "../openduck_in_solvespace/split_parts/part_16.obj"
                              ]
                            }
                            DEF left_leg_4 CadShape {
                              url [
                                "../openduck_in_solvespace/split_parts/part_17.obj"
                              ]
                            }
                            DEF left_fifth_motor CadShape {
                              url [
                                "../openduck_in_solvespace/split_parts/part_20.obj"
                              ]
                            }
                            DEF fifth_motor HingeJoint {
                              jointParameters HingeJointParameters {
                                anchor 0.15503 0.01792 -0.14001
                              }
                              device [
                                RotationalMotor {
                                  name "t5"
                                  maxVelocity 300
                                  maxTorque 300
                                }
                                PositionSensor {
                                  name "t5_sensor"
                                }
                              ]
                              endPoint DEF left_sole Solid {
                                children [
                                  DEF left_sole CadShape {
                                    url [
                                      "../openduck_in_solvespace/split_parts/part_22.obj"
                                    ]
                                  }
                                ]
                                boundingObject Pose {
                                  translation 0.1368 0.011 -0.15
                                  children [
                                    Box {
                                      size 0.0469 0.09241 0.0413
                                    }
                                  ]
                                }
                              }
                            }
                          ]
                          boundingObject Pose {
                            translation 0.13333 0.016 -0.10105
                            children [
                              Box {
                                size 0.04775 0.028 0.09874
                              }
                            ]
                          }
                          physics Physics {
                          }
                        }
                      }
                    ]
                    boundingObject Pose {
                      translation 0.13333 0.015 -0.02
                      children [
                        Box {
                          size 0.04775 0.028 0.09874
                        }
                      ]
                    }
                    physics Physics {
                    }
                  }
                }
              ]
              name "left_shoulder"
              boundingObject DEF left_shoulder Pose {
                translation 0.105 -0.001 0.0175
                children [
                  Box {
                    size 0.08943 0.047 0.028
                  }
                ]
              }
              physics Physics {
              }
            }
          }
        ]
        name "left_second_motor"
        boundingObject DEF left_second_motor Pose {
          translation 0.07 0 0.02961
          children [
            Box {
              size 0.02472 0.029 0.04522
            }
          ]
        }
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0 0 0.05422
      }
      device [
        RotationalMotor {
          name "rt1"
          maxVelocity 300
          maxTorque 300
        }
        PositionSensor {
          name "rt1_sensor"
        }
      ]
      endPoint DEF right_second_motor Solid {
        children [
          DEF right_second_motor CadShape {
            url [
              "../openduck_in_solvespace/split_parts/part_3.obj"
            ]
          }
          DEF right_shoulder HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 1 0
              anchor 0 0.01905 0.0175
            }
            device [
              RotationalMotor {
                name "rt2"
                maxVelocity 300
                maxTorque 300
              }
              PositionSensor {
                name "rt2_sensor"
              }
            ]
            endPoint Solid {
              rotation 0 1 0 0
              children [
                DEF right_shoulder CadShape {
                  url [
                    "../openduck_in_solvespace/split_parts/part_7.obj"
                  ]
                }
                DEF right_third_motor CadShape {
                  url [
                    "../openduck_in_solvespace/split_parts/part_9.obj"
                  ]
                }
                HingeJoint {
                  jointParameters HingeJointParameters {
                    anchor -0.08314 0.01464 0.01652
                  }
                  device [
                    RotationalMotor {
                      name "rt3"
                      maxVelocity 300
                      maxTorque 300
                    }
                    PositionSensor {
                      name "rt3_sensor"
                    }
                  ]
                  endPoint DEF right_upper Solid {
                    rotation 1 0 0 0
                    children [
                      DEF right_upper_1 CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_12.obj"
                        ]
                      }
                      DEF right_upper_2 CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_13.obj"
                        ]
                      }
                      DEF right_fourth_motor CadShape {
                        url [
                          "../openduck_in_solvespace/split_parts/part_15.obj"
                        ]
                      }
                      HingeJoint {
                        jointParameters HingeJointParameters {
                          anchor -0.08414 0.01628 -0.06175
                        }
                        device [
                          RotationalMotor {
                            name "rt4"
                            maxVelocity 300
                            maxTorque 300
                          }
                          PositionSensor {
                            name "rt4_sensor"
                          }
                        ]
                        endPoint DEF right_lower Solid {
                          rotation 1 0 0 0
                          children [
                            DEF right_lower_1 Shape {
                              appearance Appearance {
                                material Material {
                                }
                              }
                              geometry Mesh {
                                url [
                                  "../openduck_in_solvespace/split_parts/part_18.obj"
                                ]
                              }
                            }
                            DEF right_lower_2 Shape {
                              appearance Appearance {
                                material Material {
                                }
                              }
                              geometry Mesh {
                                url [
                                  "../openduck_in_solvespace/split_parts/part_19.obj"
                                ]
                              }
                            }
                            DEF right_fifth_motor Shape {
                              appearance Appearance {
                                material Material {
                                }
                              }
                              geometry Mesh {
                                url [
                                  "../openduck_in_solvespace/split_parts/part_21.obj"
                                ]
                              }
                            }
                            HingeJoint {
                              jointParameters HingeJointParameters {
                                anchor -0.08504 0.01792 -0.14001
                              }
                              device [
                                RotationalMotor {
                                  name "rt5"
                                  maxVelocity 300
                                  maxTorque 300
                                }
                                PositionSensor {
                                  name "rt5_sensor"
                                }
                              ]
                              endPoint DEF right_sole Solid {
                                rotation 1 0 0 0
                                children [
                                  DEF right_sole_shape Shape {
                                    appearance Appearance {
                                      material Material {
                                      }
                                    }
                                    geometry Mesh {
                                      url [
                                        "../openduck_in_solvespace/split_parts/part_23.obj"
                                      ]
                                    }
                                  }
                                  GPS {
                                    name "rt5_gps"
                                  }
                                ]
                                boundingObject Mesh {
                                  url [
                                    "../openduck_in_solvespace/checked_repaired_simplified/part_23_repaired_simplified.obj"
                                  ]
                                }
                                physics Physics {
                                }
                              }
                            }
                            GPS {
                              name "rt4_gps"
                            }
                          ]
                          name "right_lower"
                          boundingObject Group {
                            children [
                              Mesh {
                                url [
                                  "../openduck_in_solvespace/checked_repaired_simplified/part_18_repaired_simplified.obj"
                                ]
                              }
                              Mesh {
                                url [
                                  "../openduck_in_solvespace/checked_repaired_simplified/part_19_repaired_simplified.obj"
                                ]
                              }
                              Mesh {
                                url [
                                  "../openduck_in_solvespace/checked_repaired_simplified/part_21_repaired_simplified.obj"
                                ]
                              }
                            ]
                          }
                          physics Physics {
                          }
                        }
                      }
                    ]
                    name "right_upper"
                    model "right_upper"
                    boundingObject DEF right_upper Pose {
                      translation -0.063 0.015 -0.02
                      children [
                        Box {
                          size 0.04775 0.028 0.09874
                        }
                      ]
                    }
                    physics Physics {
                    }
                  }
                }
              ]
              name "right_shoulder"
              boundingObject DEF right_shoulder Pose {
                translation -0.035 -0.001 0.0175
                children [
                  Box {
                    size 0.08943 0.047 0.028
                  }
                ]
              }
              physics Physics {
              }
            }
          }
        ]
        name "right_second_motor"
        boundingObject DEF right_second_motor Pose {
          translation 0 0 0.0296
          children [
            Box {
              size 0.02472 0.029 0.04522
            }
          ]
        }
        physics Physics {
        }
      }
    }
  ]
  boundingObject DEF virtual_robot_shape Pose {
    translation 0.035 0 0.036
    children [
      Box {
        size 0.001 0.001 0.001
      }
    ]
  }
  physics Physics {
  }
  controller "walk_5_step"
  selfCollision TRUE
}
